Openrazer daemon cleanup patches

whot opened the pull request:
A bunch of patches to clean up the code a bit and make it easier to read. For the most part, this just moves things around into helper functions. The functionality changes are:
- assuming the root user is in plugdev
- ignoring screen saver monitor initialization failure

This is just the first set of patches, I suspect there will be more to come, but these are easy ones :)

whot committed:
daemon: survive failure to open screen saver

If we want to run the daemon on the system bus, we don't have an X connection.

whot committed:
daemon: s/logging_level/log_level/

if we use 'log_dir' we should use 'log_level' for consistency

whot committed:
daemon: move existing process check to a helper function

whot committed:
daemon: move creation of logger to helper function

whot committed:
daemon: move logger initialization up

So we have the logger available everywhere

whot committed:
daemon: move check for plugdev group to helper function

whot committed:
daemon: always assume that root is in plugdev

Because, well, it's the root user

whot committed:
daemon: if the 'plugdev' group lookup failed, the user isn't in it

If we cannot look up the group, the user cannot be a member of the group

whot committed:
daemon: move check for plugdev group up

No point starting a dbus daemon if we're about to quit because of our group
membership

whot committed:
daemon: move udev monitor creation into a helper function

whot committed:
daemon: move screensaver init to helper function

whot committed:
daemon: use 'BUS_NAME', not 'BUS_PATH'

Let's use the correct technical terms here

whot committed:
daemon: make the dbus method initialization easier to parse

whot committed:
daemon: remove a couple of needless checks

The defaults guaranteed these were always set anyway, we can just set those
normally in the dictionary

whot committed:
daemon: only add files ending in .py to the module import list

vim creates .foo.py.swp files while a file is open which causes the daemon to
fail with confusing error messages:
ModuleNotFoundError: No module named 'openrazer_daemon.hardware.'

whot committed:
daemon: remove unnecessary os.path.exists() check

os.path.isdir() won't ever return true if it doesn't exist

whot committed:
daemon: if the run_dir exists but is not a directory, fail early

whot committed:
daemon: create the run_dir if it doesn't exist

If a run_dir is defined, but it doesn't exist yet, just create it. We already
fail for a run_dir that exists and is a file, so we can ignore that bit now.

z3ntu commented on daemon/openrazer_daemon/daemon.py:
```diff
@@ -194,22 +194,21 @@ def __init__(self, verbose=False, log_dir=None, console_log=False, run_dir=None,
+
+        for m in methods:
+            self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
+            self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
```
Not just easier to parse, also easier to read 😃

whot committed:
daemon: don't check for existence before os.makedirs

We're making the directories anyway, so let python handle the errors. Bonus
point: we now handle the case where the target path exists but is a file

whot committed:
daemon: add the __name__ check for __main__

No real changes, just means it's really only run when being run as script

whot committed:
daemon: defer making directories until we need them

Move the run_dir and log_dir into the daemon proper, we shouldn't create them
unless we actually start the daemon. Same goes for the example config file, it
should not be created until we're sure we want to start the daemon.

whot committed:
daemon: fail if the given config file does not exist

If the user specifies a config file it must exist.
